src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1

```diff
@@ -52,165 +138,215 @@ function Set-OSServerPerformanceTunning
-            SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost" -Name "sections['webLimits'].allowDefinition" -Value "Everywhere"
-            SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost/sites/site[@name='Default Web Site']" -Name "Limits" -Value @{MaxConnections = 4294967295}
-        }
-        catch
+        if (-not $SkipIISUploadSizeLimitsConfig)
         {
-            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error configuring IIS for unlimited connections"
-            WriteNonTerminalError -Message "Error configuring IIS for unlimited connections"
+            $MaxAllowedContentLength = $AdvancedConfigurations.Sections.IISUploadSizeLimitsConfig.NewMaxAllowedContentLength
 
-            return
-        }
+            if (-not $MaxAllowedContentLength)
+            {
+                $MaxAllowedContentLength = $OSPerfTuningMaxAllowedContentLength
+            }
 
-        # Configure IIS Application Pools
-        $DefaultWebSiteApps = $(Get-WebApplication -Site "Default Web Site").Path
+            # Configure IIS requests limits (Server Level)
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring IIS upload size limits (maxAllowedContentLength = $MaxAllowedContentLength)"
+            try
+            {
+                SetWebConfigurationProperty -PSPath "MACHINE/WEBROOT/APPHOST" -Filter "system.webServer/security/requestFiltering/requestLimits" -Name "maxAllowedContentLength" -Value $MaxAllowedContentLength
+            }
+            catch
+            {
+                LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error configuring IIS upload size limits"
+                WriteNonTerminalError -Message "Error configuring IIS upload size limits"
 
-        foreach ($Config in $OSIISConfig)
-        {
-            # Reset array at each loop
-            $interestingApps = @()
+                return
+            }
+        }
 
-            # Build an array with all matching Apps.
-            foreach ($appMatchPattern in $($Config.Match))
+        if (-not $SkipIISConnectionsConfig)
+        {
+            # Configure IIS for unlimited connections. (Default Web Site) - This should not be needed cause IIS defaults to maximum.
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring IIS for unlimited connections (MaxConnections = $OSPerfTuningMaxConnections)"
+            try
             {
-                $interestingApps += $DefaultWebSiteApps | Where-Object -FilterScript {$_ -like $appMatchPattern}
+                SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost" -Name "sections['webLimits'].OverrideModeDefault" -Value "Allow"
```

Comment:
(again, not so sure about these) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

Note: This maps to the registry path/value in src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1